### ISSUE TITLE
Iron Halo, Storm Shield and Boarding Shield not providing HP bonus when equipped

### ISSUE THREAD

**Issue by renatosilvatimebel** [ZAB909/ChapterMaster#190]
This is the captain of my X company, he has 95HP:

![image](https://github.com/ZAB909/ChapterMaster/assets/69794358/d8317870-6dbe-4ced-85b5-1222d08280e4)

it remains with 95HP even that i equip any of Halo, SS and BS:

I think we have 2 issues here, one is the displayed value not being updated and the second one is the calculation itself specifically to the Iron Halo.

https://github.com/ZAB909/ChapterMaster/blob/baf62dbb15596cc3a2c293c5c5584072aec31199/scripts/scr_battle_roster/scr_battle_roster.gml#L383C18-L383C18

code is trying to find "Iron Halo" string in `deploying_unit.wep2`

shouldnt it be `deploying_unit.gear` instead?

I think the UI is not displaying the right value, but i also believe that Iron Halo is not giving its HP bonus at all in the HP calculation:
`if (deploying_unit.wep2[cooh, va] = "Iron Halo") then targ.marine_hp[targ.men] += 20;`

![image](https://github.com/ZAB909/ChapterMaster/assets/69794358/7f6e4fb5-a91c-45aa-bb43-fa095a039632)

**Comment by GalacticChimp** [comment]
I'll have some time in a few hours, so I'll take a look

**Comment by GalacticChimp** [comment]
yeah, I checked the code and it's like you say. My first idea was to add and remove buffs on equip/dequip, but I like your idea with equipment structs more

**Comment by Blogaugis** [comment]
I personally find boarding and storm shields giving HP as weird.

But If You want to have it - could it be possible to add a marine stat bonuses to equipment?
Like in this case, making the boarding and storm shields give a bonus to constitution of the marine?
It would also allow more interesting equipment in terms of mods, which tweak the stats of those who wear it.
Oh, and artifacts - these definitely should be able to provide stat modifiers.
